Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -776,14 +827,14 @@ extension String: BSONValue {
 
     public func encode(to storage: DocumentStorage, forKey key: String) throws {
         guard bson_append_utf8(storage.pointer, key, Int32(key.count), self, Int32(self.count)) else {
-            throw bsonEncodeError(value: self, forKey: key)
+            throw bsonTooLargeError(value: self, forKey: key)
         }
     }
 
     public static func from(iterator iter: DocumentIterator) throws -> String {
         var length: UInt32 = 0
         guard let strValue = bson_iter_utf8(&iter.iter, &length) else {
```

Comment:
same question as that I had about `bson_iter_symbol`. 
also, looking at docs for `bson_iter_utf8`, it says: "It is suggested that the caller validate the content is valid UTF-8 before using this in other places. That can be done by calling `bson_utf8_validate()` or validating the underlying `bson_t` before iterating it."
maybe we should do that here? (or we could open a ticket and defer it to when we add document validation later on.)